src/NServiceBus.Core/Transports/MessageContext.cs

```diff
@@ -21,20 +21,20 @@ public class MessageContext
         /// It also allows the transport to communicate to the pipeline to abort if possible. Transports should check if the token
         /// has been aborted after invoking the pipeline and roll back the message accordingly.
         /// </param>
-        /// <param name="context">Any context that the transport wants to be available on the pipeline.</param>
-        public MessageContext(string messageId, Dictionary<string, string> headers, byte[] body, TransportTransaction transportTransaction, CancellationTokenSource receiveCancellationTokenSource, ContextBag context)
+        /// <param name="extensions">Any extensions that the transport wants to be available on the pipeline.</param>
```

Comment:
Perhaps "A <see cref="ContextBag" /> which can be used to extend the current object." like used in #4468